### PR TITLE
feat: adding adr for data-test ids

### PR DESCRIPTION
**Pull request by BillyFigueroa**
## Description
Proposing an ADR for data-testid's

## Related Issue

- #183

### CODE REVIEW

File: adr/0002-testing-with-data-testids.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+## Context
+
+Front-end testing, it is often necessary to select and interact with elements on the web page to validate functionality, whether you are running unit tests or integration/ end-to-end tests. There are many different ways you can select elements, including CSS selectors, IDs, and attributes like data-testid. The choice of method impacts the maintainability and reliability of the tests
+
+When using a framework, the included locators can be used, and should be by default (i.e. getByRole, getByText, getByPlaceholder). For cases where those do not work, rather than trying to use the structure of the html to locate elements, we should use data-testid's because the framework we are using for testing provides a getByTestId locator
+
+## Decision
+
+The decision is to use the data-testid attribute for selecting elements during front-end testing when one of the default test framework selectors/ locators do not apply
```
Comment (UncleGedd) [inline adr/0002-testing-with-data-testids.md:15]: This sounds reasonable to me but defer to @jeff-mccoy

**Review by jeff-mccoy — APPROVED** [review]
Reasonable to me--but we should make sure to have contributor guidance for this and watch for it ok PRs

File: adr/0002-testing-with-data-testids.md (hunk 1 above)
Comment (BillyFigueroa) [reply adr/0002-testing-with-data-testids.md:15]: > This sounds reasonable to me but defer to @jeff-mccoy

Copy. Created #237 for this

**Comment by BillyFigueroa** [comment]
> Reasonable to me--but we should make sure to have contributor guidance for this and watch for it ok PRs



> Reasonable to me--but we should make sure to have contributor guidance for this and watch for it ok PRs

Copy. Created #237 for this

**Pull request merged**